orttraining/orttraining/eager/ort_aten.cpp

```diff
@@ -682,6 +682,68 @@ at::Tensor& out) {
   return out;
 }
 
+void copy(onnxruntime::ORTInvoker& invoker,
+          const onnxruntime::Tensor& src_tensor, onnxruntime::Tensor& dst_tensor){
+  auto& ort_ep = invoker.GetCurrentExecutionProvider();
+  ORT_THROW_IF_ERROR(ort_ep.GetDataTransfer()->CopyTensor(src_tensor, dst_tensor));
+}
+
+const at::Tensor& resize_(
+    const at::Tensor& self,
+    at::IntArrayRef size,
+    c10::optional<at::MemoryFormat> optional_memory_format) {
+  // TODO: handle named tensor...
+  // TODO: handle optional_memory_format
+  // TODO: handle existing values when resizing non-empty tensor
+
+  ORT_LOG_FN(self, size, optional_memory_format);
+  assert_tensor_supported(self);
+
+  // If self is already desired size, then return early
+  if (self.sizes() == size) {
+    return self;
+  }
+
+  auto& invoker = GetORTInvoker(self.device());
+
+  auto self_ort_value = create_ort_value(invoker, self);
+  auto* self_ort_tensor = self_ort_value.GetMutable<onnxruntime::Tensor>();
+
+  // Create new ORTValue to replace the existing ORTValue on the at::tensor
+  // object.
+  // REVIEW: Should updating the OrtValue be a method on ORTTensorImpl?
+  //         Or possibly should ORTValue / ORTTensor support resize of underlying
+  //         storage?
+  OrtValue updated_ort_value;
+  onnxruntime::Tensor::InitOrtValue(self_ort_tensor->DataType(), onnxruntime::TensorShape(size.vec()),
+        invoker.GetCurrentExecutionProvider().GetAllocator(0, OrtMemTypeDefault), updated_ort_value);
+
+  // TODO: Copy over values from original tensor to the new updated tensor
+  //       If new tensor is larger than existing tensor, copy over existing
+  //       values, leaving new memory uninitialized
+  //       If new tensor is smaller then existing tensor, copy over the range
+  //       of values that fit in new tensor (truncate existing values).
+
+  auto* self_updated_ort_tensor = updated_ort_value.GetMutable<onnxruntime::Tensor>();
+  // If new size is larger, then copy over old values
+  if (self_updated_ort_tensor->SizeInBytes() > self_ort_tensor->SizeInBytes()) {
+
+    // Currently, it is not possible to copy sub elements between tensors of different sizes
+    // as a workaround, create a new tensor (of the same size as the original self tensor)
+    // using the buffer of the larger updated tensor.
+    onnxruntime::Tensor tmpTensor{
```

Comment:
@souptc - Is this what you had in mind for creating a temporary tensor on the existing buffer, to copy between two tensors of different sizes?